docs/patched_pdoc.py

```diff
@@ -20,93 +19,86 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
+"""A script that patches some pdoc functionality before calling it."""
+import datetime
+import os
+import pathlib
+import sys
 
-import json
-import os.path as path
-import re
-
-import pdoc
-from pdoc import cli
-
-
-def _patched_generate_lunr_search(modules, index_docstrings, template_config):
-    # This will only be called once due to how we generate the documentation, so we can ignore the rest
-    assert len(modules) == 1, "expected only 1 module to be generated, got more"
-    top_module = modules[0]
-
-    def trim_docstring(docstring):
-        return re.sub(
-            r"""
-            \s+[-=~]{3,}\s+|       # title underlines
-            ^[ \t]*[`~]{3,}\w*$|   # code blocks
-            \s*[`#*]+\s*|          # common markdown chars
-            \s*([^\w\d_>])\1\s*|   # sequences of punct of the same kind
-            \s*</?\w*[^>]*>\s*     # simple HTML tags
-            \s+                    # whitespace sequences
-        """,
-            " ",
-            docstring,
-            flags=re.VERBOSE | re.MULTILINE,
+import minify_html
+import sphobjinv
+from pdoc import __main__ as pdoc_main
+from pdoc import doc as pdoc_doc
+from pdoc import render as pdoc_render
+from pdoc.render import env as pdoc_env
+
+sys.path.append(os.getcwd())
+
+import hikari
+
+pdoc_html_module = pdoc_render.html_module
+
+
+def patched_html_module(*args, **kwargs) -> str:
+    # We patch this to minify the HTML output before passing writing to the file
+    output = pdoc_html_module(*args, **kwargs)
+
+    return minify_html.minify(output, minify_js=True, minify_css=True)
+
+
+pdoc_render.html_module = patched_html_module
+
+# '-o' is the flag to provide the output dir. If it wasn't provided, we don't output the inventory
```

Comment:
Idea: create a similar hook to that of search to generate when in interactive mode?